jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/AbstractMetaDataConnection.java

```diff
@@ -0,0 +1,72 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.server;
+
+import java.net.SocketAddress;
+import java.util.concurrent.Executor;
+
+import org.eclipse.jetty.io.AbstractConnection;
+import org.eclipse.jetty.io.Connection;
+import org.eclipse.jetty.io.EndPoint;
+
+/**
+ * An {@link AbstractConnection} that also implements {@link ConnectionMetaData} with fixed
+ * local and remote addresses.
+ */
+public abstract class AbstractMetaDataConnection extends AbstractConnection implements ConnectionMetaData
+{
+    private final Connector _connector;
+    private final HttpConfiguration _httpConfiguration;
+    private final SocketAddress _localSocketAddress;
+    private final SocketAddress _remoteSocketAddress;
+
+    public AbstractMetaDataConnection(Connector connector, EndPoint endPoint, Executor executor, HttpConfiguration httpConfiguration)
```

Comment:
No need to pass the `Executor` as it is always taken from the `Connector`.
Also, I'd like to have longer lifecycle components first, so `(Connector, HttpConfiguration, EndPoint)` would be best.